lib/utils.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+const { crc32 } = require('crc');
+const { zip, gzip } = require('compressing');
+
+const FS_EXT = 'f';
+
+const getFilepath = (...parts) => `${path.join(...parts)}.${FS_EXT}`;
+
+const mkdirp = (dir, cb) => {
+  fs.access(dir, fs.constants.F_OK, err => {
+    if (err && err.code === 'ENOENT') {
+      mkdirp(path.dirname(dir), err => {
+        if (err) cb(err);
+        else fs.mkdir(dir, cb);
+      });
+    } else {
+      cb(err);
+    }
+  });
+};
+
+const computeHash = (data, checksum) => {
+  switch (checksum) {
```

Comment:
I'd prefer to have an object that contains hash functions:

```js
const hashers = {
  CRC32: (data) => crc32(data).toString(16),
  SHA256: (data) => crypto.createHash('sha256').update(data).digest('hex'),
};
```

and then use it instead of switch:
```js
const hash = hashers[checksum];
if (!hash) {
  throw new TypeError();
}
return hasher(data);
```

WDYT @SemenchenkoVitaliy @lundibundi?